Create LBHD.lsl

MalefactorIX opened the pull request:
LBHD is based on the same idea of adding directional modifiers to LBA damage sources.

It is to be considered an extension of LBA slim and features minimal anti-grief capability.

This method uses angles to determine the frontal, rear, and side position of an object and has so far proven reliable against a variety of damage sources including avatar-bound, rezzed, and impact sources.

MalefactorIX committed:
Create LBHD.lsl

MalefactorIX committed:
Update LBHD.lsl

MalefactorIX committed:
Update LBHD.lsl

MalefactorIX committed:
Update LBHD.lsl

HadetTheUndying commented:
Just a side note, I recommend adding a version number to the script so it's easier to keep track of changes once it hits Inworld. For example LBA 2.07 had that damage freezing bug, and we can easily spot it so long as people named their scripts properly thankfully. I don't see any glaring issues with the code, looks pretty clean, will need to test it this evening.

MalefactorIX committed:
Cleaning/optimization pass

That should take care of all the deprecated code and formatting that was left over from the original script.

MalefactorIX committed:
Renamed to make purpose/version easier to find

MalefactorIX committed:
Fixed incorrect desc, Added top/bottom support

Version 1.2 Update adds modifiers and threshold for top/bottom hits and corrects the modifiers being set incorrectly in the description.

HadetTheUndying commented:
Okay so going through this inworld. I agree with what @ThunderRahja mentioned in Discord earlier. I can or you can rename some of the variables like `tpos` and `trot` To something that's more easily searchable in the script like `targetRot` or something because searching the other names means you can't replace all or anything because of things like llGetPos and all that fun stuff.

ThunderRahja commented on LBHD, Directional Armor v1.2.lsl:
llList2Key

ThunderRahja commented on LBHD, Directional Armor v1.2.lsl:
llList2Float

ThunderRahja commented on LBHD, Directional Armor v1.2.lsl:
Replace `(key)llList2String` with `llList2Key`.

ThunderRahja commented on LBHD, Directional Armor v1.2.lsl:
Replace `(float)llList2String` with `llList2Float`.

ThunderRahja commented on LBHD, Directional Armor v1.2.lsl:
This section adds a lengthy string to the object description. Consider combining all of the parameters into one instead of adding one per "modifier", or removing them entirely. There's no tangible method for scripts to use them on the fly, and scripts that attempt to are unlikely to accurately predict damage after modifiers due to the nature of SL physics and unpredictable script delays.

MalefactorIX replied:
That is mainly there so people can inspect them and make sure the vehicle/object is compliant with rules regarding damage modifiers. 
Merged it into a single string on startup. Damage sources were not intended to use these as the modifiers themselves since they are already built into the received damage process. This way every weapon which uses LBA wouldn't need an update to take advantage of directional modifiers.

ThunderRahja commented on LBHD, Directional Armor v1.2.lsl:
This conditional statement is commented out. Its following code block needs to also be commented or entirely removed.

MalefactorIX replied:
That condition is there in case people want to force line of sight check on damage sources. Some people take issue with that idea which is why it's commented out, but can still be re-enabled. Pos and TargetPos are passed to the damage process regardless of whether or not line-of-sight is performed.

ThunderRahja reviewed (changes requested):
Regarding Hadet's comment on this pull request earlier, if the variables and syntax are to be cleaned, the rest of this repository's scripts are due for a cleanup as well. I found several errors in this code that need to be fixed before implementation.

HadetTheUndying commented:
The whole repo needs a revisit when Dread and I have time I agree, but that's a discussion for another time, unless someone else is willing to put in the work. Thanks for reviewing this.

MalefactorIX committed:
Optimization Pass and Variable Adjustment

1. Damage modifiers are now stored on boot/load instead of being generated every time the object takes damage.
2. Renamed variables such as tpos and trot to targetPos and targetRot
3. Startup functions are now a global function; boot()
4. Changed some llList2* functions around

MalefactorIX committed:
Rename LBHD, Directional Armor v1.2.lsl to LBHD, Directional Armor v1.2.1.lsl

MalefactorIX committed:
1.2.2 Clean up and threshold QOL

Fixed various syntax errors and added threshold variables for front and back angles.

MalefactorIX committed:
Syntax fix, ATCAP fix

1. Fixed syntax on healing function.
2. Fixed ATCAP not actually having an effect (oops)
3. Fixed missing comma is description CSV

HadetTheUndying commented:
I'm pretty satisfied with how this looks. I want @Krutchen to please look at it tonight, and I will merge on Monday if he has no problems. I haven't had a lot of chance to test it yet, but I don't see any major issues, you've changed the few things Thunder pointed out.

ThunderRahja commented:
Looks a lot better. I can't find any pressing issues after a cursory examination.
